tests/examples/solvemathproblems.spec.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2021-Present The Serverless Workflow Specification Authors
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import * as fs from "fs";
+import {actionBuilder, foreachstateBuilder, functionBuilder, statedatafilterBuilder, workflowBuilder} from '../../src';
+
+
+describe("solvemathproblems workflow example", () => {
+	
+	
+	it('should generate Workflow object', function () {
+		
+		const workflow = workflowBuilder()
+			.id("solvemathproblems")
+			.version("1.0")
+			.name("Solve Math Problems Workflow")
+			.description("Solve math problems")
+			.start("Solve")
+			.functions([
+				functionBuilder()
+					.name("solveMathExpressionFunction")
+					.operation("http://myapis.org/mapthapis.json#solveExpression")
+					.build(),
+			])
+			.states([
+				foreachstateBuilder()
+					.name("Solve")
+					.inputCollection("${ .expressions }")
+					.iterationParam("singleexpression")
+					.outputCollection("${ .results }")
+					.actions([
+						actionBuilder()
+							.functionRef({
+								refName: "solveMathExpressionFunction",
+								arguments: {
+									"expression": "${ .singleexpression }",
+								},
+							})
+							.build(),
+					])
+					.stateDataFilter(
+						statedatafilterBuilder()
+							.output("${ .results }")
+							.build(),
+					)
+					.end(true)
+					.build(),
+			])
+			.build();
+		
+		
+		const expected = JSON.parse(fs.readFileSync("./tests/examples/solvemathproblems.json")
+			.toLocaleString()) as any;
```

Comment:
Why do you use `Buffer.toLocaleString()` instead of a more "classic" way, is there any specific reason?
```typescript
const expected = JSON.parse(fs.readFileSync("./tests/examples/solvemathproblems.json", "utf-8")) as any;
```

Futhermore, a little remark/improvement (imo), I think it's better to use the `require/import` form so the json will be included in the output directory (doing away with the necessity to fiddle with relative paths):
```typescript
// at the top level
import expected from './solvemathproblems.json';
```

Reply:
with the classic way you mean ?
```

    fs.readFile('xxx.txt', 'utf8' , (err, data) => {
         //...
      })
```

Reply:
No, that's the async way. I might be mistaken, I should double check but I think `fs.readFileSync('file')` will return a Buffer, thus your use of `toLocaleString()` I guess (or maybe there is some other reason?). Using `fs.readFileSync('file','utf8')` I think returns a string directly but I might be wrong.

Reply:
get it now, thank you